library/core/src/fmt/mod.rs

```diff
@@ -231,6 +244,243 @@ impl<W: Write + ?Sized> Write for &mut W {
+    /// down to this many characters and that truncated value is emitted with
+    /// proper fill, alignment and width if those parameters are set.
+    /// - For integral types, this is ignored.
+    /// - For floating-point types, this indicates how many digits after the
+    /// decimal point should be printed.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn precision(&mut self, precision: Option<usize>) -> &mut Self {
+        self.precision = precision;
+        self
+    }
+    /// Specifies whether the [`Debug`] trait should use lower-/upper-case
+    /// hexadecimal or normal integers
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn debug_as_hex(&mut self, debug_as_hex: Option<DebugAsHex>) -> &mut Self {
+        self.debug_as_hex = debug_as_hex;
+        self
+    }
+
+    /// Returns the current sign (the `+` or the `-` flag).
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_sign(&self) -> Option<Sign> {
+        self.sign
+    }
+    /// Returns the current `0` flag.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_sign_aware_zero_pad(&self) -> bool {
+        self.sign_aware_zero_pad
+    }
+    /// Returns the current `#` flag.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_alternate(&self) -> bool {
+        self.alternate
+    }
+    /// Returns the current fill character.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_fill(&self) -> char {
+        self.fill
+    }
+    /// Returns the current alignment.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_align(&self) -> Option<Alignment> {
+        self.align
+    }
+    /// Returns the current width.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_width(&self) -> Option<usize> {
+        self.width
+    }
+    /// Returns the current precision.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_precision(&self) -> Option<usize> {
+        self.precision
+    }
+    /// Returns the current precision.
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn get_debug_as_hex(&self) -> Option<DebugAsHex> {
+        self.debug_as_hex
+    }
+
+    /// Creates a [`Formatter`] that writes its output to the given [`Write`] trait.
+    ///
+    /// You may alternatively use [`Formatter::new()`].
+    #[unstable(feature = "formatting_options", issue = "118117")]
+    pub fn create_formatter<'a>(self, write: &'a mut (dyn Write + 'a)) -> Formatter<'a> {
+        Formatter { options: self, buf: write }
+    }
+
+    #[doc(hidden)]
+    #[unstable(feature = "fmt_internals", reason = "internal to standard library", issue = "none")]
+    /// Flags for formatting
+    pub fn flags(&mut self, flags: u32) {
```

Comment:
Good question, I don't remember the reason anymore. Maybe this was needed in some previous version of my solution, or this is just a copy+paste error I made or something. Either way: I also don't see why this would be necessary, so I went ahead and made both functions private.

Reply:
The reason came back to me (or rather, the failing CI showed me). The `formatting_options_flags` test is in `library/core/tests`, so external to the `core` sources, but needs access to the `flags`/`get_flags` methods.

I put the test there because [there aren't any other unit tests inside `core/src/fmt/`](https://github.com/rust-lang/rust/blob/a4472498d7e88041f6206faf4503eb1f246fd427/library/core/src/fmt/mod.rs#L2587-L2589), and I wanted to keep all tests in one place. However, the reason given in the link above doesn't really apply to this test – it would also be possible to pull this test inside the crate as a unit test, thereby not needing `flags`/`get_flags` to be `pub` anymore, but also splitting the `fmt` tests into two places. I don't have a strong opinion here, if you like this better than keeping `flags`/`get_flags` public but hidden, I'd be happy to change that. I'll revert the commit making `flags` and `get_flags` for now.

Reply:
> The `formatting_options_flags` test is in `library/core/tests`, so external to the `core` sources, but needs access to the `flags`/`get_flags` methods.

Ah yes, regular `#[test]`s don't work in libcore. Same thing here: https://github.com/rust-lang/rust/blob/cce6a6e22e715bd74455f2560a956ab920c3a914/library/core/src/num/dec2flt/mod.rs#L72-L76

Can you add a clarifying comment or update the `reason =` like in the example above?

Reply:
Good point, done.